plugins/router/technicolor_tc8715d

```diff
@@ -0,0 +1,195 @@
+        self.downstream_channels = 0
+
+        # Number of upstream channels.
+        self.upstream_channels = 0
+
+        self.downstream_SNRs = list()
+        self.downstream_powers = list()
+        self.upstream_powers = list()
+        self.unerrored_codewords = list()
+        self.correctable_codewords = list()
+        self.uncorrectable_codewords = list()
+
+    def handle_data(self, data):
+        data = data.strip()
+        if data != "":
+            self.signaldatapage.append(data)
+
+    def process(self):
+        # Delete the junk before the statistics start. This list
+        # should start with 'Downstream' after this deletion.
+        del self.signaldatapage[0:119]
+
+        index_positions = [i for i, x in enumerate(self.signaldatapage) if x == "Index"]
+        lock_status_positions = [
+            i for i, x in enumerate(self.signaldatapage) if x == "Lock Status"
+        ]
+        self.downstream_channels = lock_status_positions[0] - index_positions[0] - 1
+        self.upstream_channels = lock_status_positions[1] - index_positions[1] - 1
+
+        self.downstream_SNRs = self.signaldatapage[
+            6 + 3 * self.downstream_channels : 22 + 3 * self.downstream_channels
+        ]
+        self.downstream_SNRs = [x.split()[0] for x in self.downstream_SNRs]
+
+        self.downstream_powers = self.signaldatapage[
+            7 + 4 * self.downstream_channels : 23 + 4 * self.downstream_channels
+        ]
+        self.downstream_powers = [x.split()[0] for x in self.downstream_powers]
+
+        self.upstream_powers = self.signaldatapage[
+            15
+            + 6 * self.downstream_channels
+            + 4 * self.upstream_channels : 15
+            + 6 * self.downstream_channels
+            + 5 * self.upstream_channels
+        ]
+        self.upstream_powers = [x.split()[0] for x in self.upstream_powers]
+
+        self.unerrored_codewords = self.signaldatapage[
+            19
+            + 6 * self.downstream_channels
+            + 7 * self.upstream_channels : 19
+            + 7 * self.downstream_channels
+            + 7 * self.upstream_channels
+        ]
+
+        self.correctable_codewords = self.signaldatapage[
+            20
+            + 7 * self.downstream_channels
+            + 7 * self.upstream_channels : 20
+            + 8 * self.downstream_channels
+            + 7 * self.upstream_channels
+        ]
+
+        self.uncorrectable_codewords = self.signaldatapage[
+            21
+            + 8 * self.downstream_channels
+            + 7 * self.upstream_channels : 21
+            + 9 * self.downstream_channels
+            + 7 * self.upstream_channels
+        ]
+
+
+if len(sys.argv) != 2 or sys.argv[1] != "config":
+    print(
+        "Error: plugin designed for the dirtyconfig protocol, must be run with the config argument"
+    )
+    sys.exit(1)
+
+parser = TechnicolorHTMLParser()
+
+for line in urllib.request.urlopen("http://192.168.100.1/vendor_network.asp"):
```

Comment:
It would be great, if the IP would be configurable.

Reply:
It doesn't need to be configurable because the cable modem is always at this special address. It cannot be changed.